### DISCUSSION TITLE
JOOQ parser adds and true at the end of delete statement for BigQuery dialect

### DISCUSSION THREAD

**Discussion by apptio-speravali** [jOOQ/jOOQ#16868] [Q&A] [answered]
Jooq version: 3.19.8

SQL statement :  
```
delete from `db1.deleteTargetTable`
where col1 = 2;
```
Code:
```
val jooqConfig = new DefaultConfiguration
jooqConfig.setSQLDialect(SQLDialect.BIGQUERY)
DSL.using(jooqConfig)
      .parser
      .parseStatement(
        statement
      )
```

Result ;
```
delete from `db1.deleteTargetTable`
where (
  COL1 = 2
  and true
)
```

expected:
```
delete from `db1.deleteTargetTable`
where (
  COL1 = 2
)
```

**Comment by lukaseder** [comment, marked as answer]
Thanks for your message.

This has already been discovered and fixed in jOOQ 3.20 as part of a major refactoring of `DELETE` and `UPDATE` statement support due to these issues:

- #13412
- #13425
- #16631
- #16632
- #16633
- #16634 

The unnecessary `and true` predicate hasn't been extracted into a separate issue as it was mostly cosmetic, and the fix was dependent on the above complex changes, which is why the fix wasn't backported.

The reason why we generate such a predicate at all is because ClickHouse and BigQuery have mandatory `WHERE` clauses in `UPDATE` and `DELETE`.

Do you have a use-case where this matters, where you'd wish for a backport? Would upgrading to a 3.20.0-SNAPSHOT version be acceptable in your case?

**Reply by apptio-speravali** [reply]
We have UnitTests where it fails the SQL generated statement. I can wait until 3.20 is officially released.

**Reply by lukaseder** [reply]
I'll look into a backport: https://github.com/jOOQ/jOOQ/issues/16882. Currently fixing a few other BigQuery related issues, so there's some momentum...
